### PR TITLE
Update decoders to parse float and return null on error

### LINKED ISSUE #432: Parse/decode `numeric` columns as float/int
Hello, is there any way to parse numeric column values as `int` to `float`? We are currently getting all values from our DB that are `numeric` as `string`.

I saw they library uses some decoders internally, is there any way to customize what fields one wants to parse as number to string?

### PR DESCRIPTION
**Pull request by bombillazo**
Fixes #432

### Breaking Changes
- On value decode error, log error and return `null` (previously bubbled the error up)
- Parse `Float4` to number (previously returned as string)

### Fixes
- Format all project files using `deno fmt`
- Update PG geometric type decoders with improved parse checks and error messaging

### CODE REVIEW

File: tests/data_types_test.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { generateSimpleClientTest } from "./helpers.ts";
 import type {
   Box,
   Circle,
-  Float4,
+  // Float4,
```
Comment (bartlomieju) [inline tests/data_types_test.ts:7]: Should we remove it if not used?

File: tests/decode_test.ts
Comment (bartlomieju) [inline tests/decode_test.ts:1]: Great tests 🎉

**Review by bartlomieju — APPROVED** [review]
LGTM

File: tests/data_types_test.ts (hunk 1 above)
Comment (bombillazo) [reply tests/data_types_test.ts:7]: Eventually, I'm thinking we might use it if we let users add custom parsers  #239 

The way I envision the API for parsers is to have two main parsing modes:
- 'string' : all values are returned as string, and the user has to take care of parsing (`Float4` would be used in this case)
- 'auto' : deno-postgres parses the data into JS objects (as many as possibly implemented [here](https://github.com/denodrivers/postgres/issues/184), non-implemented parsers will return strings)

Both modes are overridable with some API to pass custom parser **_per_** oid type, giving users complete power over how they want any data type to be parsed by the library.

Comment (bartlomieju) [reply tests/data_types_test.ts:7]: That makes sense, thanks!

**Pull request merged**